zephyr/CMakeLists.txt

```diff
@@ -1,3 +1,30 @@
+set(PROJECT_VERSION_MAJOR 2)
+set(PROJECT_VERSION_MINOR 2)
+set(PROJECT_VERSION_PATCH 0)
+set(PROJECT_VERSION 2.2.0)
+# these must be moved to Kconfig
+option(UCLIENT_PROFILE_DISCOVERY "Enable discovery profile." OFF)
+option(UCLIENT_PROFILE_UDP "Enable UDP transport." OFF)
+option(UCLIENT_PROFILE_TCP "Enable TCP transport." OFF)
+option(UCLIENT_PROFILE_CAN "Enable CAN FD transport." OFF)
+option(UCLIENT_PROFILE_SERIAL "Enable Serial transport." OFF)
+option(UCLIENT_PROFILE_STREAM_FRAMING "Enable stream framing protocol." ON)
+option(UCLIENT_PROFILE_CUSTOM_TRANSPORT "Enable Custom transport." ON)
+
+set(UCLIENT_MAX_OUTPUT_BEST_EFFORT_STREAMS 1 CACHE STRING "Set the maximum number of output best-effort streams for session.")
+set(UCLIENT_MAX_OUTPUT_RELIABLE_STREAMS 1 CACHE STRING "Set the maximum number of output reliable streams for session.")
+set(UCLIENT_MAX_INPUT_BEST_EFFORT_STREAMS 1 CACHE STRING "Set the maximum number of input best-effort streams for session.")
+set(UCLIENT_MAX_INPUT_RELIABLE_STREAMS 1 CACHE STRING "Set the maximum number of input reliable streams for session.")
+set(UCLIENT_MAX_SESSION_CONNECTION_ATTEMPTS 10 CACHE STRING "Set the number of connection attemps.")
+set(UCLIENT_MIN_SESSION_CONNECTION_INTERVAL 1000 CACHE STRING "Set the connection interval in milliseconds.")
+set(UCLIENT_MIN_HEARTBEAT_TIME_INTERVAL 100 CACHE STRING "Set the time interval between heartbeats in milliseconds.")
+set(UCLIENT_UDP_TRANSPORT_MTU 512 CACHE STRING "Set the UDP transport MTU.")
+set(UCLIENT_TCP_TRANSPORT_MTU 512 CACHE STRING "Set the TCP transport MTU.")
+set(UCLIENT_SERIAL_TRANSPORT_MTU 512 CACHE STRING "Set the Serial transport MTU.")
+set(UCLIENT_CUSTOM_TRANSPORT_MTU 512 CACHE STRING "Set the Custom transport MTU.")
+
+set(UCLIENT_PLATFORM_ZEPHYR ON)
+
```

Comment:
These variables are only for `configure_file()` down below.  And I'd like to replace them with Kconfig if needed in near future.

I'm sure you moved them up here because you wanted to use `UCLIENT_PROFILE_STREAM_FRAMING`, no?  If so, Would you mind replacing `UCLIENT_PROFILE_STREAM_FRAMING` with Kconfig, just like you did for `MICRO_XRCE_DDS_CLIENT_CUSTOM`?  We don't need the generator expression for the following line:
```
$<$<BOOL:${UCLIENT_PROFILE_STREAM_FRAMING}>:${ZEPHYR_CURRENT_MODULE_DIR}/src/c/profile/transport/stream_framing/stream_framing_protocol.c>
```

Reply:
I've re-ordered the CMakeLists.txt file.
Also tested that we do not need to enable the stream framing protocol for now - We use UDP in our Custom Transport provider, which maintains packet boundaries below the MTU